### ISSUE TITLE
[PP-OCRv6][Performance] High CPU inference latency and low CPU utilization - need Small model accuracy with lower latency

### ISSUE THREAD

**Issue by Nandhu910** [PaddlePaddle/PaddleOCR#18320]
### 🔎 Search before asking

- [x] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [x] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [x] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

## Problem

We are trying to achieve low OCR latency while maintaining good
accuracy for invoices and table documents.

We initially ran PaddleOCR with:

2 vCPU / 4 GB RAM

and then increased the infrastructure to:

4 vCPU / 8 GB RAM

However, increasing the resources provided only around 0.2 seconds
latency improvement.

More importantly, during OCR inference the container does not appear
to fully utilize the available CPU.

Although approximately 3.5 vCPU is allocated to the pod, observed CPU
consumption is typically only around 1-1.5 vCPU.

Memory consumption is also only approximately 1-1.5 GB.

Therefore, increasing the pod from 2 vCPU to approximately 4 vCPU has
not resulted in proportional inference performance improvement.

Example profiling:

OCR_INFERENCE_MS=7549
OCR_PARSING_MS=25
TOTAL_REQUEST_MS=8115

OCR_INFERENCE_MS=6981
OCR_PARSING_MS=14
TOTAL_REQUEST_MS=7017

OCR_INFERENCE_MS=8871
OCR_PARSING_MS=42
TOTAL_REQUEST_MS=8937

OCR_INFERENCE_MS=8555
OCR_PARSING_MS=20
TOTAL_REQUEST_MS=9300

Almost all request latency is inside PaddleOCR inference.
Application parsing takes only a few milliseconds.

## Environment

We are running PaddleOCR as an OCR API on AWS EKS.

- PaddleOCR: 3.7.0
- PaddlePaddle: 3.2.0
- PaddleX: 3.7.x
- Python: 3.10
- Device: CPU
- AWS Region: eu-north-1
- EC2 instance: c6i.xlarge
- Instance resources: 4 vCPU / 8 GiB RAM
- Pod CPU request/limit: 3500m
- Pod memory: 5-6 GiB
- Architecture: linux/amd64

Runtime configuration:

PADDLEOCR_CPU_THREADS=3
PADDLEOCR_ENABLE_MKLDNN=true
OMP_NUM_THREADS=3
MKL_NUM_THREADS=3
OPENBLAS_NUM_THREADS=1
NUMEXPR_NUM_THREADS=1
PADDLEOCR_DET_LIMIT=1024

## PP-OCRv6 Tiny vs Small

We tested Tiny models to improve latency:

text_detection_model_name="PP-OCRv6_tiny_det"
text_recognition_model_name="PP-OCRv6_tiny_rec"

Latency improved significantly.

However, Tiny causes accuracy problems with small/narrow text in
invoice tables.

For example, our invoice contains a narrow column:

DG

with values:

1
1
1
1
...

Some of these values are not captured when using Tiny.

We therefore tested:

text_detection_model_name="PP-OCRv6_small_det"
text_recognition_model_name="PP-OCRv6_tiny_rec"

and increased:

text_det_limit_side_len=1280

However, the small-text accuracy issue still occurs.

Using:

text_detection_model_name="PP-OCRv6_small_det"
text_recognition_model_name="PP-OCRv6_small_rec"

provides better accuracy, but inference latency becomes significantly
higher.

Therefore we currently have this trade-off:

Tiny:
Good latency, insufficient accuracy for small invoice/table text.

Small:
Good accuracy, but significantly higher CPU latency.

**Comment by Heavygrass** [comment]
There's some computationally **heavy** piece of code that is **NOT parallel** for **CPU-only** run (even for purely local run).
`PaddleOCRVL(device="cpu", cpu_threads=11)`

After this output it can't use more then 1 core (at least for PaddleOCR-VL, but the issue is likely common):
```
(DELETED PART)venv_paddleocr/lib/python3.12/site-packages/paddle/tensor/creation.py:1152: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach(), rather than paddle.to_tensor(sourceTensor).
  return tensor(
```

**Comment by dongyikuan919** [comment]
I traced the current inference defaults in `main` and found two knobs that are easy to miss for this CPU case:

- `paddleocr/_constants.py` defaults `cpu_threads` to 10 and enables MKL-DNN.
- The OCR pipeline leaves `text_detection_batch_size`, `textline_orientation_batch_size`, and `text_recognition_batch_size` unset, so the underlying PaddleX pipeline chooses its own defaults rather than exposing a workload-aware CPU batch size.

For the reported 4-vCPU deployment, could we benchmark the same invoice with an explicit matrix such as:

```python
PaddleOCR(
    device="cpu",
    cpu_threads=3,
    enable_mkldnn=True,
    text_detection_batch_size=2,
    textline_orientation_batch_size=8,
    text_recognition_batch_size=16,
)
```

and compare it with the current defaults while recording per-stage latency and CPU utilization? If explicit batch sizes improve throughput without changing recognition results, exposing/documenting those defaults for CPU deployments would address the latency/utilization part independently from the Tiny-vs-Small accuracy trade-off.

The model-size trade-off itself still needs a benchmark on the narrow invoice column, so I would avoid changing the default model until that accuracy/latency data is available.
